Tests/HubTests/ConfigTests.swift

```diff
@@ -0,0 +1,430 @@
+//
+//  ConfigTests.swift
+//  swift-transformers
+//
+//  Created by Piotr Kowalczuk on 13.03.25.
+//
+
+import Foundation
+import Jinja
+import XCTest
+
+@testable import Hub
+
+class ConfigGeneralTests: XCTestCase {
+    func testHashable() throws {
+        let testCases: [(Config.Data, Config.Data)] = [
+            (Config.Data.integer(1), Config.Data.integer(2)),
+            (Config.Data.string("a"), Config.Data.string("2")),
+            (Config.Data.boolean(true), Config.Data.string("T")),
+            (Config.Data.boolean(true), Config.Data.boolean(false)),
+            (Config.Data.floating(1.1), Config.Data.floating(1.1000001)),
+            (Config.Data.token((1, "a")), Config.Data.token((1, "b"))),
+            (Config.Data.token((1, "a")), Config.Data.token((2, "a"))),
+            (Config.Data.dictionary(["1": Config()]), Config.Data.dictionary(["1": 1])),
+            (Config.Data.dictionary(["1": 10]), Config.Data.dictionary(["2": 10])),
+            (Config.Data.array(["1", "2"]), Config.Data.array(["1", "3"])),
+            (Config.Data.array([1, 2]), Config.Data.array([2, 1])),
+            (Config.Data.array([true, false]), Config.Data.array([true, true])),
+        ]
+
+        for (lhs, rhs) in testCases {
+            var lhsh = Hasher()
+            var rhsh = Hasher()
+
+            lhs.hash(into: &lhsh)
+            rhs.hash(into: &rhsh)
+
+            XCTAssertNotEqual(lhsh.finalize(), rhsh.finalize())
+        }
+    }
+}
+
+class ConfigAsLiteralTests: XCTestCase {
+    func testStringLiteral() throws {
+        let cfg: Config = "test"
+        XCTAssertEqual(cfg, "test")
+    }
+
+    func testIntegerLiteral() throws {
+        let cfg: Config = 678
+        XCTAssertEqual(cfg, 678)
+    }
+
+    func testBooleanLiteral() throws {
+        let cfg: Config = true
+        XCTAssertEqual(cfg, true)
+    }
+
+    func testFloatLiteral() throws {
+        let cfg: Config = 1.1
+        XCTAssertEqual(cfg, 1.1)
+    }
+
+    func testDictionaryLiteral() throws {
+        let cfg: Config = ["key": 1.1]
+        XCTAssertEqual(cfg["key"].floating(or: 0), 1.1)
+    }
+
+    func testArrayLiteral() throws {
+        let cfg: Config = [1.1, 1.2]
+        XCTAssertEqual(cfg[0], 1.1)
+        XCTAssertEqual(cfg[1], 1.2)
+    }
+}
+
+class ConfigAccessorsTests: XCTestCase {
+    func testKeySubscript() throws {
+        let cfg: Config = ["key": 1.1]
+
+        XCTAssertEqual(cfg["key"], 1.1)
+        XCTAssertTrue(cfg["non_existent"].isNull())
+        XCTAssertTrue(cfg[1].isNull())
+    }
+
+    func testIndexSubscript() throws {
+        let cfg: Config = [1, 2, 3, 4]
+
+        XCTAssertEqual(cfg[1], 2)
+        XCTAssertTrue(cfg[99].isNull())
+        XCTAssertTrue(cfg[-1].isNull())
+    }
+
```

Comment:
Very nice test suite! Would you mind adding something simple like the following, even if only for documentation purposes?

```suggestion

    func testDynamicLookup() throws {
        let cfg: Config = ["model_type": "bert"]

        XCTAssertEqual(cfg["model_type"], "bert")
        XCTAssertEqual(cfg.modelType, "bert")
        XCTAssertEqual(cfg.model_type, "bert")
        XCTAssertTrue(cfg.unknown_key.isNull())
    }
```